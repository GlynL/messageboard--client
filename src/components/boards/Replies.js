import React from "react";
import { connect } from "react-redux";
import NewReply from "./NewReply";
import "../../styles/replies.css";

export const Replies = props => {
  const renderReplies = () =>
    props.thread.replies.map(reply => (
      <li className="replies__list-item" key={reply._id}>
        <p>{reply.text}</p>
      </li>
    ));
  return (
    <div className="replies">
      <h1>Replies</h1>
      <ul className="replies__list">{renderReplies()}</ul>
      <NewReply />
    </div>
  );
};

const mapStateToProps = state => ({ thread: state.threads.activeThread });

export default connect(mapStateToProps)(Replies);
