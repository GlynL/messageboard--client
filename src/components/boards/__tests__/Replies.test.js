import React from "react";
import { shallow } from "enzyme";
import { Replies } from "../Replies";

it("matches snapshot", () => {
  const wrapper = shallow(
    <Replies
      replies={[
        { _id: 1, text: "mock reply 1" },
        { _id: 2, text: "mock reply 2" }
      ]}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
