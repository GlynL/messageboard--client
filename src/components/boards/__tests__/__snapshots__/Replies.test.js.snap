// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="replies"
>
  <h1>
    Replies
  </h1>
  <ul
    className="replies__list"
  >
    <li
      className="replies__list-item"
      key="1"
    >
      <p>
        mock reply 1
      </p>
    </li>
    <li
      className="replies__list-item"
      key="2"
    >
      <p>
        mock reply 2
      </p>
    </li>
  </ul>
  <withRouter(Connect(NewReply)) />
</div>
`;
